accepted/future-releases/records/records-feature-specification.md

```diff
@@ -337,13 +347,42 @@ var c = cond ? a : b; // c has type `Record`.
 
 The greatest lower bound of records with different shapes is `Never`.
 
-### Type inference and promotion
+### Type inference
+
+As usual, we define type inference for record expressions with respect to a
+context type schema which is determined by the surrounding context of the
+inferred expression.  Given a type schema `K` and a record expression `E` of the
+general form `(e0, ..., en, d0 : t0, ..., dm : tm)`, then inference proceeds as
+follows.
+
+If `K` is a record type schema of the form `(K0, ..., Kn, {d0 : P0, ...., dm :
+Pm})`, then:
```

Comment:
Is it necessary for the implementations to perform this sorting when normalizing types?  If so, it's probably worth a short mention in non-normative text, just so that the implementation teams don't forget to do so.

Reply:
FWIW, it was more convenient for the analyzer to always ensure that named fields are sorted when we construct `RecordTypeImpl`. And positional fields are separated from named.

Reply:
> Is it necessary for the implementations to perform this sorting when normalizing types?

It is not necessary, ordering and position of named fields is always irrelevant, just as they are for named parameters for function types already.  I will add commentary to this effect, since it's causing confusion.

Reply:
I've added commentary to this effect to all relevant files.